packages/block-editor/src/components/block-list/use-block-props/index.js

```diff
@@ -104,13 +108,30 @@ export function useBlockProps( props = {}, { __unstableIsHtml } = {} ) {
 				enableAnimation:
 					! isTyping() &&
 					getGlobalBlockCount() <= BLOCK_ANIMATION_THRESHOLD,
+				blockListSettings: getBlockListSettings( clientId ),
+				hasChildBlocks: getBlockOrder( clientId ).length > 0,
 			};
 		},
 		[ clientId ]
 	);
 
 	// translators: %s: Type of block (i.e. Text, Image etc)
 	const blockLabel = sprintf( __( 'Block: %s' ), blockTitle );
+	const blockDescription = hasChildBlocks
+		? sprintf(
+				// translators: 1: Block title to lowercase for better sentence structure.
+				__(
+					'Press Escape followed by Left and Right Arrows to explore child blocks of the %s block.'
+				),
+				blockTitle.toLowerCase()
+		  )
+		: sprintf(
+				// translators: 1: Block title to lowercase for better sentence structure.
+				__(
+					'Press Tab followed by Enter to add a child block of the %s block.'
+				),
+				blockTitle.toLowerCase()
+		  );
```

Comment:
I think it's also worth checking if the block list is locked before adding this description, as that would make it impossible to add blocks.

I think the locked status is stored in the block list settings. There's a `getTemplateLock` selector which can return values like `'insert'`, `'all'`, or `false`. Generally a truthy value means insertion isn't possible.

Probably another description explaining the lock might be useful in this situation, as some users might be used to adding children to blocks like group and columns, and it'll be unexpected when they can't.

@Mamaduka is currently working on the locking functionality and might be interested in this.

Reply:
Seems like this feature isn't fully implemented yet but I did add a check for it.